clients/src/main/java/org/apache/kafka/clients/admin/TopicDescription.java

```diff
@@ -30,21 +30,38 @@
     private final boolean internal;
     private final NavigableMap<Integer, TopicPartitionInfo> partitions;
 
+    /**
+     * Create an instance with the specified parameters.
+     *
+     * @param name The topic name
+     * @param internal Whether the topic is internal to Kafka
+     * @param partitions A map from partition id to its leadership and replica information
+     */
     public TopicDescription(String name, boolean internal,
                     NavigableMap<Integer, TopicPartitionInfo> partitions) {
         this.name = name;
         this.internal = internal;
         this.partitions = partitions;
     }
 
+    /**
+     * The name of the topic.
+     */
     public String name() {
         return name;
     }
 
+    /**
+     * Whether the topic is internal to Kafka. An example of an internal topic is the offsets and group management topic:
+     * __consumer_offsets.
+     */
     public boolean internal() {
         return internal;
     }
 
+    /**
+     * A map from partition id to its leadership and replica information.
```

Comment:
How about "* A map from partition id to the leadership and replica information for that partition"